### PR TITLE
fix: Use URL from window if possible

### PR DESCRIPTION
**Pull request by jamezrin**

### CODE REVIEW

**Comment by jamezrin** [comment]
@fent The url module has been deprecated, and it is encouraged to use the URL constructor 
https://nodejs.org/api/url.html#url_url_resolve_from_to

**Comment by redbrain** [comment]
LGTM! Hopefully we can push this and then make some tests.

File: src/index.ts [hunk 1]
```diff
@@ -1,12 +1,19 @@
...
 import { Item } from './parser';
 
+/**
+ * URL constructor
+ *
+ * For compatibility with browser and Node.js APIs
+ */
+const URL = typeof window !== 'undefined' ? window.URL : require('url').URL; /* global window */
+
+exports.URL = URL;
```
Comment (TimeForANinja) [inline src/index.ts:16]:
```suggestion
```
This is not needed, just a leftover from copy pasting, right?

**Comment by TimeForANinja** [comment]
done in #92 
still thanks for the pr 😉

**Pull request closed without merging**